src/models/user.model.ts

```diff
@@ -0,0 +1,147 @@
+const USER_NOT_SAVED = 'Not able to save user: '
+const USER_NOT_DELETED = 'Not able to delete user: '
+
+class User {
+  id: String | undefined
+  name: String | undefined
+  dob: Date | undefined
+  address: String | undefined
+  description: String | undefined
+  createdAt: Date | undefined
+  updateAt: Date | undefined
+  imageUrl: String | undefined
+
+  constructor(name?: String,
+    dob?: Date,
+    address?: String,
+    description?: String,
+    imageUrl?: String) {
+    const currentDate: Date = new Date()
+    this.id = uuid()
+    this.name = name
+    this.dob = dob
+    this.address = address
+    this.description = description
+    this.createdAt = currentDate
+    this.updateAt = currentDate
+    this.imageUrl = imageUrl
+  }
+  save(): Promise<this> {
+    return mapper.put(this).catch((err) => Promise.reject(USER_NOT_SAVED + err))
+  }
+
+  static update(id: String, name?: String,
+    dob?: Date,
+    address?: String,
+    description?: String,
+    imageUrl?: String): Promise<User> {
+
+    return this.find(id).then((user) => {
+      user.name = name !== undefined ? name : user.name
+      user.dob = dob !== undefined ? dob : user.dob
+      user.address = address !== undefined ? address : user.address
+      user.description = description !== undefined ? description : user.description
+      user.updateAt = new Date()
+      return user.save()
+    })
+
+  }
+  static delete(id: String): Promise<User | undefined> {
+    const tempUser: User = new User();
+    tempUser.id = id;
+    return mapper.delete(tempUser).catch((err) => Promise.reject(USER_NOT_DELETED + err))
+  }
+
+  static find(id: String): Promise<User> {
+    const tempUser: User = new User();
+    tempUser.id = id;
+    return mapper.get(tempUser).catch((err) => Promise.reject(USER_NOT_FOUND + err))
+  }
+  static async findByName(name: String, last?: any, size: Number = 2): Promise<PaginatedUsersCompose> {
+
+    const options: QueryOptions = {
+      limit: size,
+      indexName: 'name_index'
+    }
+
+    if (last) {
+      options.startKey = { id: last.id, name: last.name }
+    }
+
+    const paginator = await mapper.query(User, { name }, options).pages()
+
+    const page = await (await paginator.next()).value
+
+    return {
+      lastEvaluatedUser: paginator.lastEvaluatedKey ? { id: paginator.lastEvaluatedKey.id, name: paginator.lastEvaluatedKey.name } : undefined,
+      users: page,
+    };
+
+  }
+  static async getAll(last?: String, size: Number = 2): Promise<PaginatedUsersId> {
+
+    const options: ScanOptions = {
+      limit: size
+    }
+    if (last) {
+      options.startKey = { id: last }
+    }
+
+    const paginator = await mapper.scan(User, options).pages()
+    const page = await (await paginator.next()).value;
+
+    return {
+      lastEvaluatedId: paginator.lastEvaluatedKey ? paginator.lastEvaluatedKey.id : undefined,
+      users: page,
+    };
+  }
+
+  static async getCoordinateFromAddress(id: String): Promise<String> {
```

Comment:
1. What are the responsibilities of your `user.model.ts` file? 
1. How would you prevent the model from becoming a hard-to-maintain **God Object** as more user-related functionalities are added to the application?

Reply:
1. The responsibility of user.model.ts file is to represent the User class, his attributes and methods.
2. By grouping related functionalities in other classes. So it would become a composition pattern, where the user model will be composed from other clases which would be responsible of certain related functionality.